ensemble_md/utils/coordinate_swap.py

```diff
@@ -0,0 +1,1334 @@
+    prev_shift_atom : int
+        Which atoms have already been shifted so we don't undo what we've done
+    """
+    atom_pair = broken_pairs_init[0]
+    broken_atom = atom_pair[1]
+    if broken_atom in prev_shift_atom:
+        broken_atom = atom_pair[0]
+    Fixed = False
+    for shift_dir in product([1, -1], [1,-1], [1, -1]): #Try all combos of shift direction
+        mol.xyz[0,broken_atom,0] = mol.xyz[0,broken_atom,0] + (shift_dir[0] * box_dimensions[0]) 
+        mol.xyz[0,broken_atom,1] = mol.xyz[0,broken_atom,1] + (shift_dir[1] * box_dimensions[1])
+        mol.xyz[0,broken_atom,2] = mol.xyz[0,broken_atom,2] + (shift_dir[2] * box_dimensions[2])
+        dist_check = md.compute_distances(mol, atom_pairs=[atom_pair], periodic=False)
+        if dist_check > 0.2: #Didn't work so reverse and try again
+            mol.xyz[0,broken_atom,0] = mol.xyz[0,broken_atom,0] - (shift_dir[0] * box_dimensions[0]) 
+            mol.xyz[0,broken_atom,1] = mol.xyz[0,broken_atom,1] - (shift_dir[1] * box_dimensions[1])
+            mol.xyz[0,broken_atom,2] = mol.xyz[0,broken_atom,2] - (shift_dir[2] * box_dimensions[2])
+        else: #Yay fixed break
+            Fixed = True
+            break
+    if Fixed:
+        prev_shift_atom.append(broken_atom)
+    return mol, Fixed
+
+def check_break(mol, atom_pairs):
+    """
+    Determine whether a break is present between the atom pairs of interest
+
+    Parameters
+    ----------
+    mol : mdtraj Trajectory object
+        Trajectory to examine for breaks
+    atom_pairs : int
+        The atom number pairs which should be connected in the residue of interest
+    
+    Returns
+    -------
+    broken_pairs : int
+        which pairs of atoms have breaks across the periodic boundary
+    """
+    dist = md.compute_distances(mol, atom_pairs=atom_pairs, periodic=False)
+    broken_pairs = []
+    for i, d in enumerate(dist[0]):
+        if d > 0.2:
+            broken_pairs.append(atom_pairs[i])
+    return broken_pairs
+
+def get_miss_coord(mol_align, mol_ref, name_align, name_ref, df_atom_swap, dir, df_swap):
+    """
+    Get coordinates for the missing atoms after the conformational swap
+
+    Parameters
+    ----------
+    mol_align : mdtraj Trajectory object
+        trajectory for the molecule being aligned
+    mol_ref : mdtraj Trajectory object
+        trajectory for the reference molecule
+    name_align : str
+        resname for the molecule which is being aligned
+    name_ref : str
+        resname for the molecule which has the reference coordinates
+    df_atom_swap : pandas DataFrame
+        For each atom that is not in the common list the dataframe gives the name of the atom, the atom number, the atom element, whether the atom is switchin between dummy and real or missing, the swap direction (same as input), whether the final atom type is real or dummy 
+    dir : str
+        swapping direction
+    df_swap : pandas DataFrame
+        For each missing atoms 

```

Comment:
@ajfriedman22 the docstring for this parameter seems incomplete. What does the dataframe have?

Reply:
Added